src/git-source-provider.ts

```diff
@@ -185,12 +188,24 @@ export async function getSource(settings: IGitSourceSettings): Promise<void> {
     // LFS fetch
     // Explicit lfs-fetch to avoid slow checkout (fetches one lfs object at a time).
     // Explicit lfs fetch will fetch lfs objects in parallel.
-    if (settings.lfs) {
+    // For sparse checkouts, let `checkout` fetch the needed objects lazily.
+    if (settings.lfs && !settings.sparseCheckout) {
       core.startGroup('Fetching LFS objects')
       await git.lfsFetch(checkoutInfo.startPoint || checkoutInfo.ref)
       core.endGroup()
     }
 
+    // Sparse checkout
+    if (settings.sparseCheckout) {
```

Comment:
what will happen if `sparseCheckout` is an empty array?

Reply:
That shouldn't happen, you can see that the `length` is checked when setting up the `sparseCheckout` variable

You can find this [here](https://github.com/actions/checkout/blob/25d6c12a40fe31e2e7ea716a2facc3bbae34a8a2/src/input-helper.ts#L87)